### PR TITLE
Support history directions other than upwards (fixes #36, related to #35)

### LINKED ISSUE #36: Request: vertical (sideways) graph
Thank you for this tool, which is very useful!  I have one repo with about 20 branches, so the graph is very wide.  Would it be possible to request a mode in which arrows to parents point left, and branches are stacked vertically?  That would give me a tall, skinny image instead of a short, wide image.  Text would still be left to right.  My apologies if this has already been implemented and I missed it in the documentation.  Thank you for considering this request!

One possible implementation in the shell is:

    git-big-picture --graphviz | sed -e '1arankdir=RL;' | dot -Tpng -o graph.png

This produces the style I am looking for.

### PR DESCRIPTION
**Pull request by hartwork**
Fixes issue #36, related to pull request #35

~Sits on top of #54 (and #49) to avoid conflicts for the moment. Happy to add tests after #50 is merged.~

### CODE REVIEW

**Comment by hartwork** [comment]
Rebase and a simple test added

**Comment by hartwork** [comment]
Rebased off of #54 now.

File: README.rst [hunk 1]
```diff
@@ -182,6 +182,9 @@ Usage
       Options to control output and format
 
       -f FMT, --format FMT  set output format [svg, png, ps, pdf, ...]
+      --history-direction {downwards,leftwards,rightwards,upwards}
+                            enforce a specific direction of history on Graphviz
+                            (reverse(!) rankdir, default: "upwards" (rankdir "TB"))
```
Comment (esc) [inline README.rst:187]: I am not sure adding `reverse(!) rankdir,` here is beneficial. It only  makes sense if one if familiar with Graphviz vernacular. Perhaps it is better to hide this detail from the user?

File: test.scf [hunk 2]
```diff
@@ -120,6 +120,14 @@ provide filename and viewer and format
     $ rm file.png
     $ ls
 
+check explicit direction of history
+    $ bash -c 'set -o pipefail; for i in {down,left,right,up}wards; do git-big-picture -g --history-direction=${i} | fgrep rankdir ; done'
```
Comment (esc) [inline test.scf:124]: What is the purpose of `set -o pipefail` here? Is it to ensure that the fails if the pipe to `fgrep` fails?

**Review by esc — CHANGES_REQUESTED** [review]
I checked this locally by trying all the available options and observing the output. Looks very nice! I left one suggestion for the docs and and one question about the tests for your consideration.

File: README.rst (hunk 1 above)
Comment (hartwork) [reply README.rst:187]: Can do, I'll follow your intuition.

File: test.scf (hunk 2 above)
Comment (hartwork) [reply test.scf:124]: The idea is to make sure that the test never succeeds if git-big-picture returns an error some time after writing the rankdir line. In Bash by default in a pipe chain the exit code is that of the last command in the chain; so if the last exits with code 0 and an earlier one exits with 1, the overall result is still 0, masking that error. For comparison:

```console
# bash             -c 'false | true ; echo $?'
0

# bash -o pipefail -c 'false | true ; echo $?'
1
```

Where needed, each individual access code can be access using array `PIPESTATUS` like so:

```console
# bash -o pipefail -c 'false | true | ( exit 2 ) ; echo "<$?> [${PIPESTATUS[@]}]"'
<2> [1 0 2]
```

Comment (hartwork) [reply test.scf:124]: PS: In Bash's own words:
```console
# help set | fgrep -A2 pipefail
              pipefail     the return value of a pipeline is the status of
                           the last command to exit with a non-zero status,
                           or zero if no command exited with a non-zero status
```

**Review by esc — APPROVED** [review]
Thanks for fixing this up, ready to merge now.

**Comment by hartwork** [comment]
Awesome!

**Pull request merged**